glass/user.py

```diff
@@ -55,23 +67,32 @@ class _FitsWriter:
     Initialised with the fits object and extension name.
     """
 
-    def __init__(self, fits, ext=None) -> None:  # type: ignore[no-untyped-def]
+    def __init__(self, fits: fitsio.FITS, ext: str = "") -> None:
         """Create a new, uninitialised writer."""
         self.fits = fits
         self.ext = ext
 
-    def _append(self, data, names=None) -> None:  # type: ignore[no-untyped-def]
+    def _append(
+        self,
+        data: npt.NDArray[np.float64] | list[npt.NDArray[np.float64]],
+        names: list[str] | None = None,
+    ) -> None:
         """Write the FITS file."""
-        if self.ext is None or self.ext not in self.fits:
+        if self.ext not in self.fits:
```

Comment:
Now `self.ext` defaults to `""` then no harm in having a blank extension

Reply:
But those are different situations: a blank EXTNAME is not the same as not having an EXTNAME.

Reply:
But the function (as I understand it) will add "", which if it already has one isn't going to do anything. I can change this to

```py
if not self.ext or self.ext not in self.fits:
```

it just felt redundant

Reply:
Have changed to `if not self.ext or self.ext not in self.fits:`